acceptance-tests/src/test/java/uk/nhs/digital/ps/test/acceptance/models/DatasetBuilder.java

```diff
@@ -0,0 +1,54 @@
+package uk.nhs.digital.ps.test.acceptance.models;
+
+import java.time.Instant;
+
+public class DatasetBuilder {
```

Comment:
Other builders in this package are immutable; they follow approach where all the `with*` methods return new instance of a builder. This is achieved through the use of `PropertySetter` interface, `cloneAndAmend(final PropertySetter propertySetter)` and copy constructors (+ the default, parameterless consturctor made private).

To keep things consistent, would you mind bringing this class in line with the other builders? `PublicationSeriesBuilder` should provide sufficient example.

Reply:
Good idea, I'll update the builder. Tried to use intellijs builder generator but it creates in this format so I'll just create it manually 🙂

Reply:
:+1: